libation: 11.5.5 -> 12.3.1

emaryn opened the pull request:
Diff: https://github.com/rmcrackan/Libation/compare/v11.5.5...v12.3.1

Changelog: https://github.com/rmcrackan/Libation/releases/tag/v12.3.1


<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->
Resolves #402642

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

TomaSajt commented on pkgs/by-name/li/libation/package.nix:
```diff
@@ -45,6 +46,13 @@ buildDotnetModule rec {
   runtimeDeps = [
     # For Avalonia UI
     glew
+    xorg.libX11
+    xorg.libXrandr
+    xorg.libXi
+    xorg.libICE
+    xorg.libSM
+    xorg.libXcursor
+    xorg.libXext
```
So, could you remove those that are unneeded?
I believe ICE and SM can be removed.
If you think the other new deps are actually improve the package, then you can keep those.

Commit:
libation: 11.5.5 -> 12.3.1

Diff: https://github.com/rmcrackan/Libation/compare/v11.5.5...v12.3.1

Changelog: https://github.com/rmcrackan/Libation/releases/tag/v12.3.1

TomaSajt commented on pkgs/by-name/li/libation/package.nix:
```diff
@@ -75,10 +79,10 @@ buildDotnetModule rec {
 
   meta = {
-    changelog = "https://github.com/rmcrackan/Libation/releases/tag/${src.rev}";
+    changelog = "https://github.com/rmcrackan/Libation/releases/tag/v${version}";
```
In these cases I'd usually use `src.tag`.

I know that in the mass package migration from rev to tag, `meta.changelog` was not migrated to `src.tag`, but to whatever `src.tag` was set to, in this case `"v${version}"`.

I don't really agree that this is better.

The `releases/tag/` endpoint needs the tag, why not specify the tag directly?

Anyways: if you have strong feelings about this, fine, but otherwise I'd want to have it as `src.tag`
